File: 0977.有序数组的平方/solution.py
from typing import List


class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        i, j, k = 0, len(nums) - 1, len(nums) - 1
        result = [None] * len(nums)
        while i <= j:
            im = nums[i] ** 2
            jm = nums[j] ** 2
            if im < jm:
                result[k] = jm
                j -= 1
            else:
                result[k] = im
                i += 1
            k -= 1
        return result

File: 0977.有序数组的平方/Solution.java
public class Solution {
    public int[] sortedSquares(int[] nums) {
        int i = 0;
        int j = nums.length - 1;
        int k = nums.length - 1;
        int[] result = new int[nums.length];
        while (i <= j) {
            int im = nums[i] * nums[i];
            int jm = nums[j] * nums[j];
            if (im < jm) {
                result[k] = jm;
                j--;
            } else {
                result[k] = im;
                i++;
            }
            k--;
        }
        return result;
    }
}
